### PR TITLE
tunnelx: minor refactor and cleanup

### PR DESCRIPTION
**Pull request by drupol**
Follow up of https://github.com/NixOS/nixpkgs/pull/239739

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/gis/tunnelx/default.nix [hunk 1]
```diff
@@ -19,34 +19,35 @@ stdenv.mkDerivation (finalAttrs: {
...
 
   installPhase = ''
+    runHook preInstall
+
     mkdir -p $out/bin $out/java
     cp -r symbols Tunnel tutorials $out/java
     makeWrapper ${jre}/bin/java $out/bin/tunnelx \
       --add-flags "-cp $out/java Tunnel.MainBox" \
-      --set SURVEX_EXECUTABLE_DIR ${survex}/bin/ \
+      --set SURVEX_EXECUTABLE_DIR ${lib.getBin survex} \
```
Comment (Artturin) [inline pkgs/applications/gis/tunnelx/default.nix:41]: lib.getBin gets the bin output, so the /bin is still necessary

Comment (drupol) [reply pkgs/applications/gis/tunnelx/default.nix]: Fixed

**Comment by drupol** [comment]
@NixOS/darwin-maintainers I would like help with this derivation. I'm pretty sure it's possible to run it on Darwin.

**Comment by drupol** [comment]
From @Cu3PO42
> The log states that the Mesa package is marked as broken on Mac. The package definition references this issue: https://gitlab.freedesktop.org/mesa/mesa/-/issues/8634 which does not appear to be resolved yet.

I'll set platforms to Linux only.

**Comment by emilazy** [comment]
I do not believe Mesa should be required for this package on Linux from what I'm seeing. Investigating now though.

**Comment by drupol** [comment]
Thanks :)

**Comment by emilazy** [comment]
This package builds but I worry I may have broken the GUI components of `survex` in the process. I'm afraid I've never gone caving outside of Colossal Cave Adventure before; could you tell me how to exercise `tunnelx`'s dependencies on those binaries?

**Comment by reckenrode** [comment]
There’s also a tracking issue for Mesa on Darwin. 

https://github.com/NixOS/nixpkgs/issues/233265

**Comment by drupol** [comment]
I have no clue, maybe @goatchurchprime can help ?

**Comment by emilazy** [comment]
Okay, I source dived and checked the `survex` executables `tunnelx` uses; they all start with this patch, even the `aven` GUI. Can't guarantee that this hasn't broken *something* on the `survex` side but it's clearly an improvement over the status quo; I'm going to call this good enough:

```nix
diff --git a/pkgs/applications/gis/tunnelx/default.nix b/pkgs/applications/gis/tunnelx/default.nix
index 174796b3bf5..21cc485e752 100644
--- a/pkgs/applications/gis/tunnelx/default.nix
+++ b/pkgs/applications/gis/tunnelx/default.nix
@@ -36,9 +36,10 @@ stdenv.mkDerivation (finalAttrs: {
 
     mkdir -p $out/bin $out/java
     cp -r symbols Tunnel tutorials $out/java
+    # `SURVEX_EXECUTABLE_DIR` must include trailing slash
     makeWrapper ${jre}/bin/java $out/bin/tunnelx \
       --add-flags "-cp $out/java Tunnel.MainBox" \
-      --set SURVEX_EXECUTABLE_DIR ${lib.getBin survex}/bin \
+      --set SURVEX_EXECUTABLE_DIR ${lib.getBin survex}/bin/ \
       --set TUNNEL_USER_DIR $out/java/
 
     runHook postInstall
diff --git a/pkgs/applications/misc/survex/default.nix b/pkgs/applications/misc/survex/default.nix
index b23cd02220b..6f2098ea477 100644
--- a/pkgs/applications/misc/survex/default.nix
+++ b/pkgs/applications/misc/survex/default.nix
@@ -45,14 +45,16 @@ stdenv.mkDerivation rec {
   buildInputs = [
     ffmpeg
     glib
-    libGLU
-    mesa
     proj
     wxGTK32
   ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
     Carbon
     Cocoa
   ] ++ lib.optionals stdenv.hostPlatform.isLinux [
+    # TODO: libGLU doesn't build for macOS because of Mesa issues
+    # (#233265); is it required for anything?
+    libGLU
+    mesa
     libICE
     libX11
   ];
```

**Comment by MatthewCroughan** [comment]
Unbelievable work, completely unexpected and brilliant. Thank you so much!

**Comment by drupol** [comment]
I was feeling a little bit concerned for merging the related pr too quickly and noticing later on that I missed plenty of things to do ... And in the meantime, the Darwin maintainers improved the Darwin support :)

**Pull request merged**